### PR TITLE
LibWeb: implement navigator.maxTouchPoints

### PR DESCRIPTION
**Pull request by ChaseKnowlden**
This is needed by vercel.com to determine how many touch points there is.

For #1298

### CODE REVIEW

File: Userland/Libraries/LibWeb/HTML/Navigator.cpp [hunk 1]
```diff
@@ -101,8 +102,15 @@ JS::NonnullGCPtr<UserActivation> Navigator::user_activation()
...
-    dbgln("FIXME: Unimplemented Navigator.maxTouchPoints");
+#if defined(AK_OS_MACOS)
+    int max_touch_points = 1;
+    return max_touch_points;
+#elif defined(AK_OS_LINUX)
+    int max_touch_points = 20;
+    return max_touch_points;
+#else
     return 0;
+#endif
```
Comment (tcl3) [inline Userland/Libraries/LibWeb/HTML/Navigator.cpp:113]: I don't believe it's sufficient to return an arbitrary number based on the platform like this. The number returned should depend on the underlying hardware of the given machine. For example, Chrome returns 10 on my touchscreen laptop, but 0 on my desktop. If we can't figure out what is supported by the hardware then I think the best thing to do is to return 0, which implies that touch input is not supported.

**Review by paaspaas00 — COMMENTED** [review]
This is linux-only, would you mind adding an implementation for Mac and possibly Windows too?

**Comment by ADKaster** [comment]
>  This is linux-only, would you mind adding an implementation for Mac and possibly Windows too?


There's no sense in adding a Windows implementation, as we can't build the application for windows.

**Comment by ChaseKnowlden** [comment]
Mac computers don't even have touch screens.

**Comment by AtkinsSJ** [comment]
> Mac computers don't even have touch screens.

If you plug in a graphics tablet or other touchscreen, that should be included I think. We need to ask the OS about it.

**Comment by ChaseKnowlden** [comment]
Will cause merge conflict after #2256 is merged.

**Comment by circl-lastname** [comment]
Apparently rebasing is enough to make it work

**Pull request closed without merging**